pkg/volume/fc/fc_util.go

```diff
@@ -140,12 +192,47 @@ func searchDisk(wwns []string, lun string, io ioHandler) (string, string) {
 	return disk, dm
 }
 
+func searchDiskWWIDs(b fcDiskMounter, wwids []string, io ioHandler) (string, string) {
+	disk := ""
+	dm := ""
+
+	rescaned := false
+	// two-phase search:
+	// first phase, search existing device path, if a multipath dm is found, exit loop
+	// otherwise, in second phase, rescan scsi bus and search again, return with any findings
+	for true {
+		for _, wwid := range wwids {
+			disk, dm = findDiskWWIDs(b, wwid, io)
+			// if multipath device is found, break
+			if dm != "" {
+				break
+			}
+		}
+		// if a dm is found, exit loop
+		if rescaned || dm != "" {
+			break
+		}
+		// rescan and search again
+		// rescan scsi bus
+		scsiHostRescan(io)
+		rescaned = true
+	}
+	return disk, dm
+}
+
 func (util *FCUtil) AttachDisk(b fcDiskMounter) error {
 	devicePath := ""
 	wwns := b.wwns
 	lun := b.lun
+	wwids := b.wwids
 	io := b.io
-	disk, dm := searchDisk(wwns, lun, io)
+	var disk, dm string
+
+	if len(wwns) != 0 {
```

Comment:
can `searchDisk()` and `searchDiskWWIDs()` be consolidated, e.g. `searchDisk(b, io)`